### ISSUE TITLE
about post-scaling

### ISSUE THREAD

**Issue by daihu-ye** [TonyPod/Two-CILs#1]
Hi,I read your paper and got some questions.In your paper ,the bias term is log( pts(c)/ptr(c)) and  pts(c) is set to be 1/c, ptr(c) is simply estimated by the ratio of the sample number of class c to the total.
In my CIL task,the protype of old class is 20,the number of samples in new class is 500.Assume the base class is 50 and the class increament is 10,in the first increament phase,the bias term i got is
`tensor([ 1.6094,  1.6094,  1.6094,..... -1.6094, -1.6094, -1.6094])`
The output of the network is quite small and the max in output is 0.3,it seems the bias term is too big and cannot get reasonal output if i simply add the bias term to the output.How can i get the right bias term?
By the way,did you use the bias term in the training phase?

**Comment by TonyPod** [comment]
Hi there, thank you for your questions! 
We did not use the bias term in the training phase, and only add it in the test phase. Therefore, the training phase is not different from ordinary CNN training, except that you should make sure that the softmax cross-entropy loss is small enough (_Sec. 3.2_ in our paper).

As for your question, there might be two possibilities: 

- `tensor([ 1.6094, 1.6094, 1.6094,..... -1.6094, -1.6094, -1.6094])` is added in the training phase in your implementation (**check this first**).

- The cross-entropy is not small enough and the assumption does not hold too well (_Sec. 3.2_). If so, more epochs or more hyperparameter tuning might be needed.

Hope it could solve your problem :)

**Comment by daihu-ye** [comment]
Thanks for your response.
The bias term is used in testing phase,assume we calculate  the bias term when the first increamental phase is trained,as for
cifar-100,now C is 60(the base class is 50,the class increament is 10),the testing samples for each class is the same(100),and the total  samples is 6000.Then the bias term log( pts(c)/ptr(c)) is
 `tensor([ 0.4520,  0.4520,  0.4520, -1.1575, -1.1575, -1.1575])`
Is is right?Did i make some mistakes in computing ?

**Comment by TonyPod** [comment]
In your setting (each old class has 20 prototypes, 50 old classes; each new class has 500 samples, 10 new classes), `pts(c) = 1/60` for each class since there are 60 classes. `ptr(c) = 20/6000 = 1/300` for old classes, and `ptr(c) = 500/6000 = 1/12` for new classes. Thus, the bias term should be `tensor([ log(5), log(5), log(5), ..., -log(5), -log(5), -log(5)]` which is `tensor([ 1.6094, 1.6094, 1.6094,..... -1.6094, -1.6094, -1.6094])` just as in your first comment.

Going back to your first comment "The output of the network is quite small and the max in output is 0.3", is the output of the network small for all test samples? 
I observe that the max in output could still reach 0.99 for some samples using our codes. The "output" here means `softmax(logits + bias_term)`.

**Comment by daihu-ye** [comment]
I try to add the bias term to another code.Output i mentioned is the tensor before softmax.Softmax just change  the range of the tensor to 0-1,the max in the tensor will still be the max after softmax.So in my test code ,i just choose the max to be the predicted class.But since the output of the network is quite small (the max in output is 0.3)and the bias term will dominant the tensor if i add the bias term to the output.So how can it work?

**Comment by daihu-ye** [comment]
It seems that the output i checked is the tensor before finishing a epoch.I check the output after training 30 epoches and the output become much bigger(max is 4,even bigger),so it make sense to add the bias term.I am so sorry to make such a mistake.
I will check my code to find if there are any bugs.
Thanks for your response.

**Comment by TonyPod** [comment]
> It seems that the output i checked is the tensor before finishing a epoch.I check the output after training 30 epoches and the output become much bigger(max is 4,even bigger),so it make sense to add the bias term.I am so sorry to make such a mistake.
> I will check my code to find if there are any bugs.
> Thanks for your response.

That's cool! I was still thinking about your previous question and found that you solved it.

Keep in touch if you have further questions :)
